### PR TITLE
fix: verify validity of next and prev commit IDs

### PR DESCRIPTION
**Pull request by miguel-crespo-fdc**
If next and prev commits were invalid, they could end up crashing cd-service (i.e id = 1)

### CODE REVIEW

File: services/cd-service/pkg/repository/transformer.go [hunk 1]
```diff
@@ -562,12 +562,12 @@ func writeCommitData(ctx context.Context, sourceCommitId string, sourceMessage s
 		return GetCreateReleaseGeneralFailure(err)
 	}
 
-	if previousCommitId != "" {
+	if previousCommitId != "" && valid.SHA1CommitID(previousCommitId) {
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/transformer.go:565]: could we have a test for this change?

File: services/cd-service/pkg/repository/transformer.go [hunk 2]
```diff
@@ -101,6 +101,9 @@ func manifestDirectoryWithReleasesVersion(fs billy.Filesystem, application strin
 }
 
 func commitDirectory(fs billy.Filesystem, commit string) string {
+	if !valid.SHA1CommitID(commit) {
+		return fs.Join("commits", "")
```
Comment (miguel-crespo-fdc) [inline services/cd-service/pkg/repository/transformer.go:105]: Should we return something different? Verification seemed missing here as invalid use of this function could lead to backend crashing

Comment (sven-urbanski-freiheit-com) [reply services/cd-service/pkg/repository/transformer.go:105]: I think at this point we need to rely that the caller already checked this.
Either that or we need to return an error here, but that complicates things.

Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/transformer.go:105]:
```suggestion

```

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**